pkg/query/iter.go

```diff
@@ -325,10 +325,10 @@ func (s *dedupSeriesSet) Next() bool {
 // replica label if it exists
 func (s *dedupSeriesSet) peekLset() labels.Labels {
 	lset := s.peek.Labels()
-	if lset[len(lset)-1].Name != s.replicaLabel {
+	if _, ok := s.replicaLabels[lset[len(lset)-1].Name]; !ok {
 		return lset
 	}
-	return lset[:len(lset)-1]
+	return lset[:len(lset)-len(s.replicaLabels)]
```

Comment:
Hm this does not looks right... What if series includes only one of the replica labels? It seems like you assume that either there are none or all replica labels specified on series.

Reply:
Not sure what semantics we should have here.. Thoughts @brancz ?

Reply:
Not sure either, I'm assuming it's similar to the other one, but at least needs a comment as well.

Reply:
I would vote for any replica label not stricly all of them

Reply:
Yeah I think the correct semantic would be to only subtract the ones that occur in the label-set, but maybe I'm missing something.

Reply:
yes that makes sense will refactor.

Reply:
I added a test for this as well.
Series that include only some of the replica labels are also de-duplicated.

see this test
https://github.com/thanos-io/thanos/pull/1362/files#diff-9dfbf115cb6790c1ff2c931fb91bf9c9R492